src/service_alerts.rs

```diff
@@ -24,14 +24,40 @@ pub const FINALIZER_NAME: &str = "servicealert.cactuar.rs";
     namespaced
 )]
 pub struct ServiceAlertSpec {
-    pub common_labels: HashMap<String, String>,
+    pub common_labels: CommonLabels,
     pub deployment_name: String,
-    pub alerts: HashMap<Alerts, Vec<AlertConfig>>,
+    pub alerts: Alerts,
+}
+
+// Since the metrics are different for different protocols, we must map each Alerts enum
+// to a different expression string in prometheus land.
+// e.g.
+// REST + ErrorPercent uses the istio_requests_total         istio standard metric
+// gRPC + ErrorPercent uses the istio_request_messages_total istio standard metric
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema, PartialEq)]
+pub struct Alerts {
+    #[serde(rename = "gRPC")]
+    pub GRPC: Option<HashMap<HttpAlerts,Vec<HttpAlertConfig>>>,
+    #[serde(rename = "REST")]
+    pub REST: Option<HashMap<HttpAlerts,Vec<HttpAlertConfig>>>,
+    // TODO: Define what is needed for misc here. look into what alerts we can support first
+    // There's a problem that the alertConfig changes for potentially every different misc
+    // alert so we can't support the Vec<HttpAlertConfig> pattern like we do with gRPC and REST
+    // pub Misc: Option<HashMap<MiscAlerts, $something>>,
```

Comment:
see comment about MiscAlerts all being quite different to each other

Reply:
Not too fussed about `HttpAlerts` being named as such for now since this in draft. If the shared enum is an identical representation other than PromQL generation, it shouldn't be a problem to keep the representation shared, since we can be conditional on what object we're looking at here:
```rust
if let Some(grpc_alerts) = crd.spec.Alerts.GRPC { ... }
```

Reply:
Proposal: How many misc alert categories do we expect to have? I'm thinking it might be cleaner to just pull those up directly to the top level alongside HTTP and gRPC, ala:
```yaml
apiVersion: cactuar.rs/v1
kind: ServiceAlert
metadata:
  name: example-service-alert
spec:
  commonLabels:
    origin: cloud
    owner: bar
  deploymentName: best-service-eu
  alerts:
    HTTP:
      latencyMillisecondsP99:
        - operation: MoreThan
          value: 20
          for: 5m
          withLabels:
            severity: warning
        - operation: MoreThan
          value: 50
          for: 2m
          withLabels:
            severity: critical
    gRPC:
      errorPercent:
        - operation: MoreThan
          value: 10 # %
          for: 3m
          withLabels:
            severity: warning
    replica:
      count:
        - operation: LessThan
          value: 3
          for: 5m
          withLabels:
            severity: warning
        - operation: EqualTo
          value: 0
          for: 1m
          withLabels:
            severity: critical
```

Reply:
Bah! Trying to make this change had me really confused... The reconciler was applying a new reconcile thousands of times a second and I couldn't figure out what I'd done. Turns out I because I was generating a unique ID every single time a reconcile took place, we weren't idempotent which threw `kube.rs` for an absolute loop.

So, note for the future: _If the reconciler is running at mach 10, you've probably broken idempotency in the reconciliation loop._

Reply:
So 
```
 replica:
      count:
```

Doesn't really work because `replicaCount` is the actual metric we're tracking and there won't be a replica${somethingElse}. We could either have a `misc` bucket or start yeeting these into the root which I don't really like the sound of personally. I quite liked your "disable standard alerts" bool and wonder how far we can take that